gradebookng/tool/src/webapp/scripts/gradebook-gbgrade-table.js

```diff
@@ -2764,6 +2764,14 @@ GbGradeTable.setupKeyboardNavigation = function() {
   });
 
   GbGradeTable.instance.addHook("beforeKeyDown", function(event) {
+    // If the wicket modal is displaying, prevent event propagation and return
+    if (document.getElementsByClassName('wicket-modal').length > 0) {
+      // The next line prevents behaviors such as keystroking tab that navigates to other handsontable cells
+      // instead of navigating to the next tabbable control in the modal.
+      event.stopImmediatePropagation();
+      return;
+    }
```

Comment:
what exactly is this doing?

Reply:
To answer that, I've posted an explainer in the form of a screencast attached to the [corresponding jira](https://sakaiproject.atlassian.net/browse/SAK-50757), titled SAK-50757-ProposedFixExplainer-20250107.mp4.

Reply:
@adrianfish does this fix make sense to you for 23.x? it seems like your fix in SAK-49442 has modified/broken this modal behavior in 23.x